enclave/src/test_proxy.rs

```diff
@@ -0,0 +1,128 @@
+use crate::polkadex;
+use crate::polkadex::PolkadexAccountsStorage;
+use codec::Encode;
+use sgx_rand::{Rng, SeedableRng, StdRng};
+use sgx_tstd::collections::HashMap;
+use sgx_types::{sgx_status_t, SgxResult};
+use sp_core::blake2_256;
+use sp_std::prelude::*;
+use std::{
+    sync::atomic::{AtomicPtr, Ordering},
+    sync::{Arc, SgxMutex},
+};
+static GLOBAL_ACCOUNTS_STORAGE: AtomicPtr<()> = AtomicPtr::new(0 as *mut ());
+
+pub fn get_dummy_map() -> HashMap<[u8; 32], Vec<[u8; 32]>> {
+    let mut hashmap: HashMap<[u8; 32], Vec<[u8; 32]>> = HashMap::new();
+    let main_account_one: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let main_account_two: [u8; 32] = Vec::from("second_account").using_encoded(blake2_256);
+    let main_account_three: [u8; 32] = Vec::from("third_account").using_encoded(blake2_256);
+    let dummy_account_one: [u8; 32] = Vec::from("first_dummy_account").using_encoded(blake2_256);
+    let dummy_account_two: [u8; 32] = Vec::from("second_dummy_account").using_encoded(blake2_256);
+    let dummy_account_three: [u8; 32] = Vec::from("third_dummy_account").using_encoded(blake2_256);
+    hashmap.insert(main_account_one, vec![dummy_account_one]);
+    hashmap.insert(main_account_two, vec![dummy_account_one, dummy_account_two]);
+    hashmap.insert(
+        main_account_three,
+        vec![dummy_account_one, dummy_account_two, dummy_account_three],
+    );
+    hashmap
+}
+
+pub fn pointer_initialize() -> sgx_status_t {
+    let mut hashmap: HashMap<[u8; 32], Vec<[u8; 32]>> = get_dummy_map();
+    let polkadex_hashmap: PolkadexAccountsStorage = PolkadexAccountsStorage::from_hashmap(hashmap);
+    let storage_ptr = Arc::new(SgxMutex::<PolkadexAccountsStorage>::new(polkadex_hashmap));
+    let ptr = Arc::into_raw(storage_ptr);
+    GLOBAL_ACCOUNTS_STORAGE.store(ptr as *mut (), Ordering::SeqCst);
+    sgx_status_t::SGX_SUCCESS
+}
+
+pub fn load_proxy_registry() -> SgxResult<&'static SgxMutex<PolkadexAccountsStorage>> {
```

Comment:
Use `load_proxy_registry` from the main file, otherwise changes within the main file might not be caught in the unit tests as the developer may forget to adapt the unit tests.

Reply:
resolved